### PR TITLE
Add Version + Release action

### PR DESCRIPTION
**Pull request by edenhaus**
Adds release action and version field, which is required for HA 2021.6+

### CODE REVIEW

**Comment by edenhaus** [comment]
@ludeeus Can you please review this PR or delegate it to the correct person?

**Comment by brianhanifin** [comment]
This looks to make this compatible with Home Assistant 2021.06? Am I right?

**Comment by edenhaus** [comment]
> This looks to make this compatible with Home Assistant 2021.06? Am I right?

Yes exactly

File: hacs.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Generate readme",
+  "iot_class": "Local Push",
```
Comment (ludeeus) [inline hacs.json:3]:
```suggestion
  "iot_class": "calculated",
```

File: custom_components/readme/manifest.json [hunk 2]
```diff
@@ -10,5 +12,5 @@
   "requirements": [
     "integrationhelper==0.2.2"
   ],
-  "homeassistant": "0.97.0"
+  "iot_class": "local_push"
```
Comment (ludeeus) [inline custom_components/readme/manifest.json:15]:
```suggestion
  "iot_class": "calculated"
```

File: custom_components/readme/const.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # Base component constants
 DOMAIN = "readme"
 DOMAIN_DATA = f"{DOMAIN}_data"
-VERSION = "0.2.1"
+INTEGRATION_VERSION = "0.2.1"
```
Comment (ludeeus) [inline custom_components/readme/const.py:5]:
```suggestion
INTEGRATION_VERSION = "0.0.0"
```

File: .github/workflows/hassfest.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: Validate with hassfest
+
+on:
+  push:
+  pull_request:
```
Comment (ludeeus) [inline .github/workflows/hassfest.yaml:5]:
```suggestion
  push:
    branches: [master]
  pull_request:
    branches: [master]
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

**Comment by ludeeus** [comment]
> Someone with enough rights please add `secrets.GITHUB_TOKEN`

This is not something you can add, that is a default special secret that GitHub injects to the action during runtime :+1:

File: custom_components/readme/manifest.json [hunk 5]
```diff
@@ -1,7 +1,9 @@
 {
   "domain": "readme",
   "name": "Generate readme",
+  "version": "v0.0.0",
```
Comment (ludeeus) [inline custom_components/readme/manifest.json:4]:
```suggestion
  "version": "0.0.0",
```

**Review by ludeeus — APPROVED** [review]
Looks good :+1: :tada:

**Review by brianhanifin — COMMENTED** [review]
The release looks to have resolved the **no 'version' key** warning in the Home Assistant 2021.05.x log. This is resolved!

File: custom_components/readme/const.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 # Base component constants
 DOMAIN = "readme"
 DOMAIN_DATA = f"{DOMAIN}_data"
-VERSION = "0.2.1"
+INTEGRATION_VERSION = "0.0.0"
```
[the remark this answers is not in the document]
Comment (ludeeus) [reply custom_components/readme/const.py:5]: It's replaced here and in the manifest

**Pull request merged**